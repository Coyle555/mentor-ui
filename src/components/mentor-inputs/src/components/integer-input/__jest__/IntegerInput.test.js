import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitForElement, cleanup, act, getByTestId } from '@testing-library/react';

import IntegerInput from '../integerInput';

test('<IntegerInput /> with no props', () => {
	const component = renderer.create(<IntegerInput />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('<IntegerInput /> with a custom className', () => {
	const component = renderer.create(<IntegerInput className="custom-class" />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('<IntegerInput /> accepts a value prop of type number', () => {
	const component = renderer.create(<IntegerInput value={3} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Integer input with a decimal value', () => {
	const component = renderer.create(<IntegerInput value={3.24} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Integer input with a non integer value', () => {
	const component = renderer.create(<IntegerInput value="test" />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Minimum value passed for integer input', () => {
	const component = renderer.create(<IntegerInput min={5} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Minimum value failed for integer input', () => {
	const component = renderer.create(<IntegerInput min={10} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('New minimum value failed for integer input', () => {
	const component = renderer.create(<IntegerInput min={0} value={5} />);
	component.update(<IntegerInput min={10} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Maximum value passed for integer input', () => {
	const component = renderer.create(<IntegerInput max={5} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Maximum value failed for integer input', () => {
	const component = renderer.create(<IntegerInput max={0} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('New maximum value failed for integer input', () => {
	const component = renderer.create(<IntegerInput max={10} value={5} />);
	component.update(<IntegerInput max={0} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
