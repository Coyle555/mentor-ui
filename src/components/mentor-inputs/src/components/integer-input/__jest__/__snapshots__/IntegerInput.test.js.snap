// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntegerInput /> accepts a value prop of type number 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="3"
/>
`;

exports[`<IntegerInput /> with a custom className 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field custom-class"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value=""
/>
`;

exports[`<IntegerInput /> with no props 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value=""
/>
`;

exports[`Integer input with a decimal value 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="3.24"
/>
`;

exports[`Integer input with a non integer value 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="test"
/>
`;

exports[`Maximum value failed for integer input 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="5"
/>
`;

exports[`Maximum value passed for integer input 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="5"
/>
`;

exports[`Minimum value failed for integer input 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="5"
/>
`;

exports[`Minimum value passed for integer input 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="5"
/>
`;

exports[`New maximum value failed for integer input 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="5"
/>
`;

exports[`New minimum value failed for integer input 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter whole number"
  type="text"
  value="5"
/>
`;
